test/e2e/accounts/test-create-snap-account.spec.js

```diff
@@ -0,0 +1,252 @@
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  unlockWallet,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL } = require('./utils');
+
+describe('Create Snap Account', function () {
+  it('create Snap account popup contains correct Snap name and snapId', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        failOnConsoleError: false,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver);
+
+        // navigate to test Snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL);
+        await driver.clickElementSafe('#connectButton');
+
+        // switch to metamask extension and click connect to start installing the snap
+        await driver.switchToWindowWithTitle('MetaMask Notification');
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        // scroll to the bottom of the page
+        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
+
+        // click the install button to install the snap
+        await driver.waitForSelector({ text: 'Install' });
+        await driver.clickElement({
+          text: 'Install',
+          tag: 'button',
+        });
+        await driver.waitForSelector({ text: 'OK' });
+        await driver.clickElement({
+          text: 'OK',
+          tag: 'button',
+        });
+
+        // move back to the Snap window to test the create account flow
+        await driver.switchToWindowWithTitle('SSK - Snap Simple Keyring');
+
+        // check the dapp connection status
+        await driver.waitForSelector({
+          css: '#snapConnected',
+          text: 'Connected',
+        });
+
+        // create new account on dapp
+        await driver.clickElement({
+          text: 'Create account',
+          tag: 'div',
+        });
+
+        await driver.clickElement({
+          text: 'Create Account',
+          tag: 'button',
+        });
+
+        // switch to metamask extension
+        await driver.switchToWindowWithTitle('MetaMask Notification');
+
+        await driver.findElement({
+          css: '[data-testid="confirmation-submit-button"]',
+          text: 'Create',
+        });
+
+        await driver.findElement({
+          css: '[data-testid="confirmation-cancel-button"]',
+          text: 'Cancel',
+        });
+
+        await driver.findElement({
+          css: '[data-testid="create-snap-account-content-description"]',
+          text: 'MetaMask Snap Simple Keyring wants to add a new Snap account to your wallet',
+        });
```

Comment:
I specifically made the opposite decision for the reasons outlined in Mark Stacey's post [here](https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549)


```
I'm noticing a common pattern in our e2e tests that I wanted to call attention to: find an element, then assert what its contents are.
This pattern inherently introduces a race condition, and should generally be avoided.
e.g. Instead of this:
const element = await driver.findElement('[data-testid="foo"]');
assert(await element.getText(), 'Bar')
Do this:
await driver.findElement({
  css: '[data-testid="foo"]',
  text: 'Bar',
});
This accomplishes the same thing, except that it's faster and doesn't have a race condition.
```